src/sentry/hybridcloud/models/outbox.py

```diff
@@ -195,7 +195,15 @@ def save(self, *args: Any, **kwargs: Any) -> None:
             )
 
         if _outbox_context.flushing_enabled:
-            transaction.on_commit(lambda: self.drain_shard(), using=router.db_for_write(type(self)))
+            try:
+                transaction.on_commit(
+                    lambda: self.drain_shard(), using=router.db_for_write(type(self))
+                )
+            except DatabaseError as e:
+                raise OutboxFlushError(
+                    f"Failed to process Outbox, {OutboxCategory(self.category).name} due to database error",
+                    self,
+                ) from e
```

Comment:
Should this try catch be *inside* `drain_shard()`? Looking at the other call sites of `drain_shard()`, all locations are already catching `OutboxFlushError` or they don't have a try/except block.

Reply:
ah yeah, sorry I didn't make the changes yesterday but yes we should be excepting within `drain_shard` since on_commit is only queueing up `drain_shard`. +1 to the question of what the purpose of `OutboxFlushError` is though, from when @GabeVillalobos  and I talked yesterday it sounded like `OutboxFlushError` is used to indicate some implementer error. If so, I think it's appropriate if we make raise something else like `OutboxDatabaseError`.